hammers/scripts/unutilized_lease_reaper.py

```diff
@@ -0,0 +1,178 @@
+# coding: utf-8
+'''
+.. code-block:: bash
+
+    unutilized-leases {info, delete}
+
+Deletes leases that do not launch any instances after a period of time
+specified by the admin.
+
+* ``info`` to just display leases or actuall delete them with ``delete``
+'''
+import argparse
+from collections import defaultdict
+from datetime import datetime, timedelta
+from pprint import pprint
+from pytz import timezone
+import sys
+
+from hammers import osapi
+from hammers.slack import Slackbot
+from hammers.notifications import _email
+from hammers.osrest import blazar, ironic, keystone
+
+DEFAULT_WARN_HOURS = 6
+DEFAULT_GRACE_HOURS = 9
+
+
+def parse_time(time):
+    time = time.split('+')[0].split('.')[0]
+    dt_fmt = '%Y-%m-%dT%H:%M:%S'
+    return datetime.strptime(time, dt_fmt).replace(tzinfo=timezone('UTC'))
+
+
+def inviolation_filter(hour):
+    now = datetime.utcnow().replace(tzinfo=timezone('UTC'))
+    threshold = now - timedelta(minutes=hour*60)
+
+    def inviolation(lease):
+        start_time = parse_time(lease['start_date'])
+
+        if start_time > threshold:
+            return False
+
+        active_nodes = any([
+            x['provision_state'] == 'active'
```

Comment:
Might want to make this `!= 'available'`, which would also catch things like the nodes stuck building (user tried to utilize lease but couldn't) or in error state (same).

Reply:
smart.